Document DynamoDB backend billing_mode option (#28672)
gravitational/teleport: docs/pages/reference/backends.mdx

    # By default, Teleport stores audit events with an AWS TTL of 1 year.
    # This value can be configured as shown below. If set to 0 seconds, TTL is disabled.
    audit_retention_period: 365d

>     # Enables On-Demand billing for the dynamodb table, if set to false, provisioned mode will be used.

Reviewer: ```suggestion
    # Enables On-Demand billing for the DynamoDB table.
    # Set to false to use provisioned capacity.
```

As an aside, defaulting a boolean to true is awkward and flies against the "make the zero value useful" paradigm.

If we want to start defaulting to on demand billing, I think we could make this clearer by either:

1. Inverting this, so you have to opt in to provisioned mode with something like `provisioned_billing: true`. (This also makes it more obvious that the capacity fields are ignored if this is not set.
2. Getting rid of booleans completely, and having `billing_mode: provisioned` or `billing_mode: on_demand`.

Reviewer 2: Right, I agree "billing_mode" enum sounds like a better fit than a boolean constant. @lxea Shall we update your PR while it's still not merged? https://github.com/gravitational/teleport/pull/27674

    # default: true

>     on_demand: true

Reviewer: ```suggestion
    on_demand_billing: true
```

    write_min_capacity: int
    write_max_capacity: int
    write_target_value: float

>     # Enables On-Demand billing

Reviewer: These settings only take effect if the table does not exist and Teleport has to create it, right?

If so, that's not clear and we should probably clarify.

    # default: true

>     on_demand: [true|false]

Reviewer: I would move this above the read/write capacity values and also mention here that the capacity values are ignored for on-demand mode.
